lib/identifier/uuid/uuid.rb

```diff
@@ -1,5 +1,17 @@
 module Identifier
   module UUID
+    MATCHER = %r{
+      [a-f0-9]{8}          # abcdef01
+      -                    #         -
+      [a-f0-9]{4}          #          abcd
+      -                    #              -
+      4[a-f0-9]{3}         #               4abc
+      -                    #                   -
+      [89aAbB][a-f0-9]{3}  #                    8abc
```

Comment:
This is a "should" vs "must" decision then. I'm okay with conditional compliance here, as the regex for all-same case would be awkward (and likely require additional ruby code).